### PR TITLE
deps(identity): update ed25519-dalek to 2.0

### LINKED ISSUE #4327: RUSTSEC-2022-0093
## Description

[RUSTSEC-2022-0093](https://rustsec.org/advisories/RUSTSEC-2022-0093) has just been issued which will make `cargo deny` fail on the CI. I tried updating `identity` and `noise` that depend on it, but we are blocked by https://github.com/mcginty/snow/issues/164

### PR DESCRIPTION
**Pull request by altonen**
## Description

Fixes #4327 for the `v0.51` release.
Related: #4337.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -55,3 +55,61 @@ members = [
...
+
+
+[patch.crates-io]
+
+# Patch away `libp2p-idnentity` in our dependency tree with the workspace version.
+# `libp2p-identity` is a leaf dependency and used within `rust-multiaddr` which is **not** part of the workspace.
+# As a result, we cannot just reference the workspace version in our crates because the types would mismatch with what
+# we import via `rust-multiaddr`.
+# This is expected to stay here until we move `libp2p-identity` to a separate repository which makes the dependency relationship more obvious.
+libp2p-identity = { path = "identity" }
```
Comment (thomaseizinger) [inline Cargo.toml]: I don't think we should introduce this in the backport. That was a feature we landed afterwards.

File: core/Cargo.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ fnv = "1.0"
 futures = { version = "0.3.28", features = ["executor", "thread-pool"] }
 futures-timer = "3"
 instant = "0.1.11"
-libp2p-identity = { version = "0.1", path = "../identity", features = ["peerid", "ed25519"] }
+libp2p-identity = { version = "0.2.3", path = "../identity", features = ["peerid", "ed25519"] }
```
Comment (thomaseizinger) [inline core/Cargo.toml]: We can't make the bump to `0.2` here, that is a breaking change and can't be released as a patch. You'll to need to patch the `0.1` source code of `libp2p-identity`.

File: identity/Cargo.toml [hunk 3]
```diff
@@ -13,8 +13,8 @@ categories = ["cryptography"]
 
 [dependencies]
 asn1_der = { version = "0.7.6", optional = true }
-bs58 = { version = "0.4.0", optional = true }
-ed25519-dalek = { version = "1.0.1", optional = true }
+bs58 = { version = "0.5.0", optional = true }
```
Comment (thomaseizinger) [inline identity/Cargo.toml]: Is this necessary?

File: identity/Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "libp2p-identity"
-version = "0.1.2"
+version = "0.2.3"
```
Comment (thomaseizinger) [inline identity/Cargo.toml]: Like I said above, this needs to be `0.1.3`.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks! We'll need to reduce this in scope, both to be as minimal as possible and for it to actually be releasable as patch.

File: identity/Cargo.toml (hunk 3 above)
Comment (altonen) [reply identity/Cargo.toml]: It wasn't, removed

File: core/Cargo.toml [hunk 5]
```diff
@@ -16,7 +16,7 @@ fnv = "1.0"
 futures = { version = "0.3.28", features = ["executor", "thread-pool"] }
 futures-timer = "3"
 instant = "0.1.11"
-libp2p-identity = { version = "0.1", path = "../identity", features = ["peerid", "ed25519"] }
+libp2p-identity = { version = "0.1.3", path = "../identity", features = ["peerid", "ed25519"] }
```
Comment (mxinden) [inline core/Cargo.toml]: Why do we need to bump the `libp2p-identity` version in `libp2p-core` or any of the other crates below? Given that we will release a patch of `libp2p-identity`, downstream consumers can simply upgrade their `Cargo.lock`. It doesn't require any intermediate crates to upgrade. Am I missing something?

Comment (altonen) [reply core/Cargo.toml]: I understood there was no downside to upgrading them to use the latest version but I can change them back

**Review by thomaseizinger — APPROVED** [review]
This looks good to me, thank you!

**Review by mxinden — APPROVED** [review]

**Comment by mxinden** [comment]
[Tagged](https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-identity-v0.1.3) and [published](https://crates.io/crates/libp2p-identity/0.1.3).

//CC @altonen

**Pull request merged**